src/core_modules/capture-core/rules/useRuleEngineFlags.js

```diff
@@ -0,0 +1,23 @@
+// @flow
+import { useEffect } from 'react';
+import { useLocationQuery } from '../utils/routing';
+import { rulesEngine } from './rulesEngine';
+
+export const useRuleEngineFlags = () => {
+    // This hook is used to set the verbose flag on the rules engine
+    // based on the verbose query param in the URL
+
+    const { verbose } = useLocationQuery();
+
+    const updateFlags = (flags) => {
+        rulesEngine.setFlags({ ...rulesEngine.getFlags(), ...flags });
+    };
+
+    useEffect(() => {
+        if (verbose === 'true') {
+            updateFlags({ verbose: true });
+        } else {
+            updateFlags({ verbose: false });
+        }
+    }, [verbose]);
```

Comment:
With this approach we can in theory run the rules engine before the verbose flag has been set, because an `useEffect` in an inner component will run before this one (in practice it will probably work because we use async functions). Did you put this in an useEffect because side-effects should not be part of the rendering, or because some part of react-router-dom is not fully initialised in the render? The preferred solution here might be to use `useLayoutEffect` (as that one is fired before `useLayout`)

Reply:
No special thoughts behind it, just choosing useEffect over useLayoutEffect by default. Switched now 👍